src/analysis/special_functions/gaussian.lean

```diff
@@ -5,18 +5,92 @@ Authors: Sébastien Gouëzel
+* `complex.Gamma_one_half_eq`: the formula `Γ (1 / 2) = √π`.
 -/
 
 noncomputable theory
 
 open real set measure_theory filter asymptotics
 open_locale real topological_space
 
+open complex (hiding exp continuous_exp abs_of_nonneg)
+notation `cexp` := complex.exp
+
+/-- If `f` is a function `ℂ → ℂ` which is continuous on a preconnected set `S`, and
+`(f z) ^ 2 = 1` for all `z ∈ S`, then either `f = 1` on `S`, or `f = -1` on `S`. -/
+lemma is_preconnected.eq_one_or_eq_neg_one_of_sq_eq {α 𝕜 : Type*}
+  [topological_space α] [field 𝕜] [topological_space 𝕜] [t1_space 𝕜]
+  {S : set α} (hS : is_preconnected S)
+  {f : α → 𝕜} (hf : continuous_on f S) (hsq : eq_on (f ^ 2) 1 S) :
+  (eq_on f 1 S) ∨ (eq_on f (-1) S) :=
+begin
+  simp_rw [eq_on, pi.one_apply, pi.pow_apply, sq_eq_one_iff] at hsq,
+  -- First deal with crazy case where `S` is empty.
+  by_cases hS : ∀ (x:α), x ∉ S,
+  { left, intros x hx,
+    exfalso, exact hS x hx, },
+  push_neg at hS,
+  choose y hy using hS,
+  suffices : ∀ (x:α), x ∈ S → f x = f y,
+  { rcases (hsq hy),
+    { left, intros z hz, rw [pi.one_apply z, ←h], exact this z hz, },
+    { right, intros z hz, rw [pi.neg_apply, pi.one_apply, ←h], exact this z hz, } },
+  refine λ x hx, hS.constant_of_maps_to hf (λ z hz, _) x y hx hy,
+  show f z ∈ ({-1, 1} : set 𝕜),
+  { exact mem_insert_iff.mpr (hsq hz).symm,  },
+  exact discrete_of_t1_of_finite,
+end
+
+/-- If `f, g` are functions `ℂ → ℂ` which are both continuous on a preconnected set `S`, and
+`(f z) ^ 2 = (g z) ^ 2` and `g z ≠ 0` hold for all `z ∈ S`, then `f = g` on `S`. -/
+lemma is_preconnected.eq_or_eq_neg_of_sq_eq {α 𝕜 : Type*}
+  [topological_space α] [field 𝕜] [topological_space 𝕜] [t1_space 𝕜]
+  [has_continuous_inv₀ 𝕜] [has_continuous_mul 𝕜]
+  {S : set α} (hS : is_preconnected S)
+  {f g : α → 𝕜} (hf : continuous_on f S) (hg : continuous_on g S)
+  (hsq : eq_on (f ^ 2) (g ^ 2) S) (hg_ne : ∀ {x:α}, x ∈ S → g x ≠ 0) :
+  (eq_on f g S) ∨ (eq_on f (-g) S) :=
+begin
+  rcases hS.eq_one_or_eq_neg_one_of_sq_eq (hf.div hg (λ z hz, hg_ne hz)) (λ x hx, _) with h | h,
+  { refine or.inl (λ x hx, _),
+    rw ←div_eq_one_iff_eq (hg_ne hx),
+    exact h hx },
+  { refine or.inr (λ x hx, _),
+    specialize h hx,
+    rwa [pi.div_apply, pi.neg_apply, pi.one_apply, div_eq_iff (hg_ne hx), neg_one_mul] at h,  },
+  { rw [pi.one_apply, div_pow, pi.div_apply, hsq hx, div_self],
+    exact pow_ne_zero _ (hg_ne hx) },
+end
+
+lemma is_preconnected.eq_of_sq_eq {α 𝕜 : Type*}
+  [topological_space α] [field 𝕜] [topological_space 𝕜] [t1_space 𝕜]
+  [has_continuous_inv₀ 𝕜] [has_continuous_mul 𝕜]
+  {S : set α} (hS : is_preconnected S)
+  {f g : α → 𝕜} (hf : continuous_on f S) (hg : continuous_on g S)
+  (hsq : eq_on (f ^ 2) (g ^ 2) S)
+  (hg_ne : ∀ {x:α}, x ∈ S → g x ≠ 0)
+  {y : α} (hy : y ∈ S) (hy' : f y = g y) : eq_on f g S :=
+λ x hx, begin
+  rcases hS.eq_or_eq_neg_of_sq_eq hf hg @hsq @hg_ne with h | h,
+  exact h hx,
```

Comment:
```suggestion
  { exact h hx },
```
When you have two goals, each one should be enclosed with braces to separate them cleanly.